keras/saving/serialization_lib_test.py

```diff
@@ -94,6 +122,13 @@ def test_builtin_layers(self):
         self.assertEqual(layer.trainable, restored.trainable)
         self.assertEqual(layer.compute_dtype, restored.compute_dtype)
 
+    def test_numpy_get_item_layer(self):
+        input = keras.layers.Input(shape=(2,))
+        layer = input[:, 1]
+        model = keras.Model(input, layer)
+        serialized, restored, reserialized = self.roundtrip(model)
+        self.assertEqual(serialized, reserialized)
```

Comment:
You can remove all the other added code by just doing e.g. 

```python
serialized_str = str(serialized).replace("(", "[").replace(")", "]")
reserialized_str = str(reserialized)
self.assertEqual(serialized_str, reserialized_str)
```

Reply:
If you prefer it, I can do that but honestly, isn't that a bit ad-hoc to fix this only for this special case? Wouldn't that problem turn up in general for future tests? Tuples being left out in `test_simple_objects()` even though they appear a lot, e.g. for all shape specifications, seemed odd to me, too.

Also, plain string manipulation feels a bit arbitrary and could hide problems when the replacement happens e.g. inside string payload where in general, you would want to detect differences. Of course, for this specific case, it's not a problem but depending on how the serialization format evolves, I am not sure, it will stay like that. Are you?

So, please confirm, you prefer the string comparison and I will adjust it.

Reply:
@fchollet Please let me know!

Reply:
It's a matter of complexity. The cost of complexity is high. The need to check object equality on one line in a test does not justify the added complexity and its cost. The alternative formulation is adhoc, but it is simple and easy to read, so its cost is almost 0.

Reply:
OK, your project, your call. I have been developing complex software for many years now, and my take is to always prefer the proper solution in contrast to ad-hoc fix-ups but of course, in a unit-test only context, it does not pay so much to spend more time debating this. I will change it probably tomorrow.

Reply:
@fchollet Actually, did you try your suggestion? It does not work here because `'shape': (None, 2)` is actually still handled as a tuple after reserialization in contrast to other things. So, I have to map both strings.